Dithering utils

wtgee opened the pull request:
These are pulled from https://github.com/AstroHuntsman/huntsman-pocs

Besides some slight code cleanup the only difference is to specify a default `pattern_offset` of 30 arcminutes.

Note that these are placed in the utils folder and are not yet a part of normal operations.

AnthonyHorton commented on pocs/utils/dither.py:
```diff
@@ -0,0 +1,95 @@
+
+        # Get n_positions from the pattern
+        ra_offsets = [pattern[count % len(pattern)][0] for count in range(n_positions)]
+        dec_offsets = [pattern[count % len(pattern)][1] for count in range(n_positions)]
```
For the sake of readability perhaps the `[count % len(pattern) for count in range(n_positions)]` should be extracted into a descriptively named utility function? I find these two lines a bit opaque as they stand, and I wrote them!

wtgee commented:
@AnthonyHorton I've added the plotting options back although changed slightly. Mostly I'm not using pyplot but instead generating an actual figure. As such I just return the figure and it is up to the user to display/save.

I also simplified it a bit so the only param to the plotting function is the generated positions. This means it does not show the given offsets as part of the legend or title.  Since the figure is returned to the user these could be added in after the fact if desired.

I've also added testing via `pytest-mpl`

wtgee commented:
I've wasted too much time on the dithering plot but did finally get the image upload of the failed images to work. Here's the output (baseline, created, diff). Needless to say I'm pretty frustrated. Note that this has the `remove_text=True` option so it should be ignoring the text but I guess that is only the title. Apparently there is some minor difference in the font style that is rending a 13% rms on the image diff. 😠 

![baseline-test_plot_dither](https://user-images.githubusercontent.com/70736/48973979-4baf1880-f0a0-11e8-8712-cb38fc4e5eb3.png)

![test_plot_dither](https://user-images.githubusercontent.com/70736/48973980-51a4f980-f0a0-11e8-91c2-cc382f8c0d1c.png)

![test_plot_dither-failed-diff](https://user-images.githubusercontent.com/70736/48973998-5669ad80-f0a0-11e8-9d67-68bdf5bed3e6.png)

wtgee commented on .travis.yml:
```diff
@@ -91,3 +91,5 @@ cache:
 after_success:
   - bash <(curl -s https://codecov.io/bash)
+after_failure:
+  - bash scripts/testing/failed-images-upload.sh ${POCS}/test_images/
```
@jamessynge when you get a chance I would appreciate your thoughts on this.
